yarn-project/acir-simulator/src/client/private_execution.ts

```diff
@@ -249,26 +249,24 @@ export class PrivateFunctionExecution {
     const argsSize = this.abi.parameters.reduce((acc, param) => acc + sizeOfType(param.type), 0);
     const contractDeploymentData = this.context.txContext.contractDeploymentData ?? ContractDeploymentData.empty();
 
-    // NOTE: PSA to anyone updating this code: within the structs, the members must be in alphabetical order, this
-    // is a current quirk in noir struct encoding, feel free to remove this note when this changes
     const fields = [
-      this.callContext.isContractDeployment,
-      this.callContext.isDelegateCall,
-      this.callContext.isStaticCall,
       this.callContext.msgSender,
-      this.callContext.portalContractAddress,
       this.callContext.storageContractAddress,
+      this.callContext.portalContractAddress,
+      this.callContext.isDelegateCall,
+      this.callContext.isStaticCall,
+      this.callContext.isContractDeployment,
+
+      this.context.historicRoots.privateDataTreeRoot,
+      this.context.historicRoots.nullifierTreeRoot,
+      this.context.historicRoots.contractTreeRoot,
+      this.context.historicRoots.l1ToL2MessagesTreeRoot,
```

Comment:
serialization now follows user-defined ordering in structs

Reply:
Should the todo not be updated then? Atop of `writeInputs` there was a todo for ordering and sorting on noir side.